apps/domewatch/dome_watch.star

```diff
@@ -0,0 +1,353 @@
+
+# Returns a small font when needed to fit text on the screen.
+def getFloorStatusFont(floor):
+    if floor["now"]["value"] == "adjourned":
+        return "tom-thumb"
+    else:
+        return "5x8"
+
+# Returns a dulled color when House is not in session.
+def getFloorStatusFontColor(floor):
+    if floor["now"]["value"] == "adjourned":
+        return "#1e4980"
+    elif floor["now"]["value"] == "house_not_in_session":
+        return "#14355c"
+    else:
+        return "#FFFFFF"
+
+def getFloorActivityFromAPI():
+    floor_cached = cache.get("floor")
+    if floor_cached != None:
+        print("Using cached floor activity")
+
+        # De-stringify the floor object.
+        floor = json.decode(floor_cached)
+    else:
+        print("Getting floor activity from API")
+        response = http.get(DOME_WATCH_API_URL + "/floor", headers = {
```

Comment:
Check out the http module's built in [cache functionality](https://discuss.tidbyt.com/t/http-caching-client/5129). Should simplify this part of the app a fair bit.